### PR TITLE
WebGPURenderer: Add timestamp queries support in WebGPU

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
This PR add support for timestamp queries to monitor WebGPU performances. (Requires Chrome 121)

In my opinion this feature should be included OOTB in the Renderer, it will be really useful to compare WebGPU vs WebGL performances and is optional (disabled by default). I will add WebGL fallback via `EXT_disjoint_timer_query_webgl2` in a second time.

The monitoring of performances can be enable with:
`new WebGPURenderer( { trackTimestamp: true } );`

Also improved the info system and honored all render calls to be async as they should (`render()` is async).
[honor async render and move to renderer await functions](https://github.com/mrdoob/three.js/pull/27597/commits/1d62772cf640892d8bccb7f29a33c6fc9ae1d1e3)

Live demo:
[https://raw.githack.com/renaudrohlinger/three.js/timestamp-queries/examples/webgpu_compute_particles.html](https://raw.githack.com/renaudrohlinger/three.js/timestamp-queries/examples/webgpu_compute_particles.html
)
Screenshot 2024-01-20 at 18 57 30

_This contribution is funded by [Utsubo](https://utsubo.com/)_

### CODE REVIEW

**Comment by sunag** [comment]
I think we could standardize async functions like `compileAsync` and do the same in `renderAsync()` and `computeAsync()`, they last are just asynchronous because they need to wait for the device to initialize.

I think we can get users used to using the suffix at first even if `renderAsync()` returns the `render()` function for now, this would make it easier for us when we change.

/cc @Mugen87

**Comment by RenaudRohlinger** [comment]
Make sense.
Added the new `Async` syntax.

Everything is now `computeAsync` and `renderAsync` and there is now some getters that simply call the async methods. For example:
```js
get render() {

return this.renderAsync;

}
```

/cc @sunag

File: examples/jsm/renderers/common/Backend.js [hunk 1]
```diff
@@ -90,6 +90,8 @@ class Backend {
 
 	// utils
 
+	resolveTimeStampAsync( renderContext, type ) { }
```
Comment (LeviPesin) [inline examples/jsm/renderers/common/Backend.js:93]: Shouldn't it be `Timestamp` instead of `TimeStamp` for consistency? /ping @RenaudRohlinger

Comment (mrdoob) [reply examples/jsm/renderers/common/Backend.js:93]: Yep! 👍

**Pull request merged**